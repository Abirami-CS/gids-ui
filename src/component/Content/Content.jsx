import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import './Content.scss';
import Landing from '../Landing/Landing';
import FirstGame from '../FirstGame/FirstGame';

const Content = () => (
  <main>
    <h1 className="heading">
      <span>Save the Server</span>
    </h1>

    <BrowserRouter>
      <Route
        render={({ location }) => {
          const { pathname } = location;
          return (
            <TransitionGroup>
              <CSSTransition
                key={pathname}
                classNames="page"
                timeout={{
                  enter: 1000,
                  exit: 1000,
                }}
              >
                <Route
                  location={location}
                  render={() => (
                    <Switch>
                      <Route exact path="/" component={Landing} />
                      <Route path="/firstgame" component={FirstGame} />
                    </Switch>
                  )}
                />
              </CSSTransition>
            </TransitionGroup>
          );
        }}
      />
    </BrowserRouter>
  </main>
);

export default Content;
