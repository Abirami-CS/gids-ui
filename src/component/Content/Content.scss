.heading {
    background: url('../../../asserts/img/dashboard.png') no-repeat center;
    text-align: center;
    height: 200px;
    line-height: 200px;
    background-size: cover;
    span {
        color: #fff;
        font-size: 55px;
        font-family: openExBold;
    }
}