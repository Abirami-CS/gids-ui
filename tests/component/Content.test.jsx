import { shallow } from 'enzyme';
import React from 'react';
import Content from '../../src/component/Content/Content';

describe('Content', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Content />);
    expect(wrapper).toMatchSnapshot();
  });
});
