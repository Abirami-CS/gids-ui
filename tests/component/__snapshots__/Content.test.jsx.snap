// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should match the snapshot 1`] = `
<main>
  <h1
    className="heading"
  >
    content
    <span>
      working
    </span>
  </h1>
</main>
`;
